services/galley/src/Galley/API/MLS/Message.hs

```diff
@@ -669,13 +671,13 @@ processExternalCommit ::
   Qualified UserId ->
   Maybe ClientId ->
   Local Data.Conversation ->
+  ConversationMLSData ->
   ClientMap ->
   Epoch ->
-  GroupId ->
   ProposalAction ->
   Maybe UpdatePath ->
   Sem r ()
-processExternalCommit qusr mSenderClient lconv cm epoch groupId action updatePath = withCommitLock groupId epoch $ do
+processExternalCommit qusr mSenderClient lconv mlsMeta cm epoch action updatePath = withCommitLock (cnvmlsGroupId mlsMeta) epoch $ do
```

Comment:
The `ConvNotFound` error constraint likely isn't needed here either.

Reply:
Unfortunately because of `checkGroup` it can't be removed